### DISCUSSION TITLE
ANT BMS no communication

### DISCUSSION THREAD

**Discussion by rudikof01** [Louisvdw/dbus-serialbattery#395] [Q&A]
Hello
I have a problem with the ANT BMS, I can't connect the raspberry pi with the bms, the leds on the rs232 convent are flashing a few times. Venus OS 2.91 and driver v.0.14.3. Does anyone have an idea what is wrong.
![ant](https://user-images.githubusercontent.com/117858685/211100456-d93f5f59-7e78-4ce6-9f61-6a75eb9b0b9b.jpg)

**Comment by Louisvdw** [comment]
Can you give a picture or description of how you wired the BMS to the USB adapter and which adapter you used?

**Reply by rudikof01** [reply]
Hello
I use an FTDI adapter. This adapter was disconnected from another ANT BMS and I saw it was connected to the original Cebro GX. I use raspberry PI 3. The LED flashes a few times, after a series of flashes, the second LED flashes several times and the whole cycle repeats after a dozen or so seconds.
![IMG_20230109_224901](https://user-images.githubusercontent.com/117858685/211415730-a1a0973a-2825-467a-9584-e57ecd8997e3.jpg)

**Reply by Louisvdw** [reply]
OK this looks like a TTL adapter. Make sure the the voltage selection is for 3.3V (although in many cases it still works on 5V also)
Then the green TX should match your connectors outmost wire as per the image below and the yellow one next to it.
![image](https://user-images.githubusercontent.com/26077665/211540536-967297e8-b70d-4e15-8c31-f007847c9618.png)

**Reply by rudikof01** [reply]
Hello
My TTL adapter is set to 3.3V. Rx and Tx are connected exactly as suggested. Maybe the BMS is not working properly, but I have no idea how to check it, I was thinking of buying a dedicated display. I'll add some screenshots.
![rp](https://user-images.githubusercontent.com/117858685/211634788-30eefe35-bfeb-4992-8f9a-582e9b5c2788.jpg)

https://user-images.githubusercontent.com/117858685/211634799-9bdeea12-6fd3-4cd0-b4bc-660f77d31c68.mp4

**Comment by rudikof01** [comment]
Hello. I tried to change the ttl converter, unfortunately it also did not work. It seems that ANT BMS works ok, I was able to connect via ESP32 and Home Assistant. I really have no idea what else to check.
![IMG_20230114_205848](https://user-images.githubusercontent.com/117858685/212494446-40590d61-e057-4e66-897b-22fa879570cb.jpg)
![IMG_20230114_205910](https://user-images.githubusercontent.com/117858685/212494447-2322e997-51f3-4c05-a692-dc3be13e6a4d.jpg)

**Comment by Louisvdw** [comment]
It's possible that you might have one of the ANT bms which use the new protocol that we still need to add 
https://github.com/Louisvdw/dbus-serialbattery/issues/262

**Reply by rudikof01** [reply]
Thank you for your time. My ANT BMS was produced 09-2022 and I uploaded the code to esp32 from this project : https://github.com/syssi/esphome-ant-bms.
I will wait with the hope that it will be possible to add the new ANT protocol to the driver as well.
Regards, you're doing a great job.
